### PR TITLE
ci: production AWS stack upgrade testing before deployment

### PR DESCRIPTION
**Pull request by imincik**
## Reference

[Code review checklist](CODING.md#Code-review-checklist)

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -8,7 +8,200 @@ on:
...
+      if: >
+        startsWith(github.ref, 'refs/heads/release-') || startsWith(github.ref, 'refs/tags/release-')
+        && github.repository == 'linz/geospatial-data-lake'
+      run: |
+        poetry run cdk bootstrap aws://unknown-account/ap-southeast-2
+        poetry run cdk deploy --all --require-approval never
+
+
+  upgrade-copy-of-prod:
+    needs: deploy-copy-of-prod
```
Comment (l0b0) [inline .github/workflows/deploy.yml:116]: Is this intentional? Doesn't this mean that we can't upgrade without running deploy *in the same pipeline* first?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -8,7 +8,200 @@ on:
...
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python: ['3.8']
```
Comment (l0b0) [inline .github/workflows/deploy.yml:120]: There is so much duplicate code here now that I think we should probably pull out a few reusable snippets. What do you think?

**Review by l0b0 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (imincik) [reply .github/workflows/deploy.yml:120]: Yes, there's a lot of duplication, but I would pull out code in separate PR

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (imincik) [reply .github/workflows/deploy.yml:116]: This is testing of prod upgrade done on prod copy in CI environment

**Pull request merged**